File: API/Entities/Uputnica.cs
namespace API.Entities
{
    public class Uputnica
    {
        public int Id { get; set; }
        public int PacijentId { get; set; }
        public Pacijent Pacijent { get; set; } = null!;
        public int DoktorId { get; set; }
        public Doktor Doktor { get; set; } = null!;
        public required string Dijagnoza { get; set; }
        public required string Opis { get; set; } 
        public required string UpucujeSe { get; set; }
        public DateTime DatumIzdavanja { get; set; } = DateTime.Now;
    }
}

File: client/src/models/Uputnica.ts

export type Uputnica = {
  id: number;
  pacijentId: number;
  doktorId: number;
  dijagnoza: string;
  opis: string;
  upucujeSe: string;
  datumIzdavanja: string;

  pacijent?: {
    id: number;
    ime: string;
    prezime: string;
  };

  doktor?: {
    id: number;
    ime: string;
    prezime: string;
  };
}

export type UputnicaDto = {
  dijagnoza: string;
  opis: string;
  upucujeSe: string;
  doktorId: number; 
}
